src/cpp/session/modules/SessionThemes.R

```diff
@@ -1164,3 +1219,12 @@
       theme[names(theme) != "url"]
    })
 })
+
+# RPC Functions ====================================================================================
+.rs.addJsonRpcHandler("get_theme_name", function(themeFile) {
+   conn <- file(themeFile)
+   lines <- readLines(conn)
+   close(conn)
```

Comment:
The first two lines should read:

    conn <- file(themeFile)
    on.exit(close(conn), add = TRUE)

Then you don't have to worry about the connection leaking if e.g. `readLines` throws an error.